### PR TITLE
fix(servers): more descriptive errors when calls fail

### PR DESCRIPTION
**Pull request by niklasad1**
Closing #775

### CODE REVIEW

File: core/src/server/helpers.rs [hunk 1]
```diff
@@ -119,8 +119,8 @@ impl MethodSink {
 				tracing::error!("Error serializing response: {:?}", err);
 
 				if err.is_io() {
-					let data = to_json_raw_value(&format!("Exceeded max limit {}", self.max_response_size)).ok();
-					let err = ErrorObject::borrowed(OVERSIZED_RESPONSE_CODE, &OVERSIZED_RESPONSE_MSG, data.as_deref());
+					let data = format!("Exceeded max limit {}", self.max_response_size);
```
Comment (niklasad1) [inline core/src/server/helpers.rs:122]: makes more sense to use `ErrrorObject::owned` when we already have to allocate a `String` here.

File: tests/tests/integration_tests.rs [hunk 2]
```diff
@@ -537,11 +537,13 @@ async fn ws_server_limit_subs_per_conn_works() {
 	let err1 = c1.subscribe::<usize>("subscribe_forever", None, "unsubscribe_forever").await;
 	let err2 = c1.subscribe::<usize>("subscribe_forever", None, "unsubscribe_forever").await;
 
+	let data = "\"Exceeded max limit 10\"";
```
Comment (jsdw) [inline tests/tests/integration_tests.rs:540]: Nit: "Exceeded max limit of 10"?

**Review by jsdw — APPROVED** [review]
Nice! just a wee grammar nit

**Review by TarikGul — APPROVED** [review]
LGTM, nice QOL update for errors

**Pull request merged**